src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/histogram.js

```diff
@@ -0,0 +1,533 @@
+         *
+         * Based on https://www.d3-graph-gallery.com/graph/interactivity_brush.html
+         * Section "Brushing for zooming".
+         */
+        const zoomIn = () => {
+            let xMin = scale.x.invert(extent[0]);
+            let xMax = scale.x.invert(extent[1]);
+
+            scale.x.domain([xMin,xMax]);
+            const dx = (extent[1]-extent[0]);
+            scale.zoom = scale.zoom * (canvas.inner.width / dx);
+
+            self.rescale(scale,axis,canvas,bars,true);
+        }
+
+        const zoomInKeyEvent = (event) => {
+            if (shortcuts.zoomIn(event)) {
+                zoomIn();
+                endBrushing();
+            }
+        };
+
+        /**
+         * Updates plot when brushing.
+         */
+        function updateChart() {
+            let selectionEvent            = d3.event.selection;
+            selectedZoomBtn.style.display = "inline-block";
+            selectedZoomBtn.addEventListener("click",zoomIn,true)
+            document.addEventListener('keydown',zoomInKeyEvent,true);
+            extent = selectionEvent;
+        }
+
+        /**
+         * Removes brush, keyboard event and zoom button when end event is captured.
+         */
+        const endBrushing = () => {
+            brushElem.call(brush.move,null);
+            selectedZoomBtn.style.display = "none";
+            selectedZoomBtn.removeEventListener("click",zoomIn,true)
+            document.removeEventListener('keydown',zoomInKeyEvent,true);
+        };
+
+        let endEvents = ['click','auxclick','contextmenu','scroll']
+        endEvents.forEach(e => document.addEventListener(e,endBrushing,false));
+    }
+
+    /**
+     * Helper function for rescaling the data points with a new scale.
+     */
+    rescale(scale, axis, canvas, bars, with_animation) {
+        const animation_duration = (with_animation ? ANIMATION_DURATION : 0.0);
+        axis.x.transition().duration(animation_duration)
+            .call(d3.axisBottom(scale.x).ticks(canvas.inner.width / X_AXIS_LABEL_WIDTH));
+        bars
+            .transition().duration(animation_duration)
+            .attr('transform',d => "translate(" + scale.x(d.x0) + "," + scale.y(d.length) + ")scale(" + scale.zoom + ",1)")
+    }
+
+    /**
+     * Create the d3 histogram for the given data.
+     */
+    createHistogram(extremesAndDeltas,canvas,svg,dataPoints,focus) {
+        let domain_x = [extremesAndDeltas.xMin - extremesAndDeltas.paddingX,
+            extremesAndDeltas.xMax + extremesAndDeltas.paddingX];
+
+        if (focus !== undefined) {
+            if (focus.x !== undefined && focus.zoom !== undefined) {
+                let padding_x = extremesAndDeltas.dx * (1 / (2 * (focus.zoom)));
+                domain_x = [focus.x - padding_x,focus.x + padding_x];
+            }
+        }
+
+        let x = d3.scaleLinear()
+            .domain(domain_x)
+            .range([0,canvas.inner.width]);
+        let xAxis = svg.append("g")
```

Comment:
You are sometimes adding nlns between lets, sometimes not. Make it consistent please. (Im ok with both)